text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,623 @@
+  --no-run`, or equivalent "smoke tests". In particular, this requirement may
+  apply if the target builds host tools, or if the tests in question provide
+  substantial value via early detection of critical problems.
+- Building the target in CI must not take substantially longer than the current
+  slowest target in CI. This requirement is subjective, to be evaluated by the
+  infrastructure team, and will take the community importance of the target
+  into account.
+- New tier 2 targets should, if at all possible, support cross-compiling. New
+  tier 2 targets should not require using the target as the host for builds,
+  even if the target supports host tools.
+- In addition to the legal requirements for all targets (specified in the tier
+  3 requirements), because a tier 2 target typically involves the Rust project
+  building and supplying various compiled binaries, incorporating the target
+  and redistributing any resulting compiled binaries (e.g. built libraries,
+  host tools if any) must not impose any onerous license requirements on any
+  members of the Rust project, including infrastructure team members and those
+  operating CI systems. This is a subjective requirement, to be evaluated by
+  the approving teams.
+  - As an exception to this, if the target's primary purpose is to build
+    components for a Free and Open Source Software (FOSS) project licensed
+    under "copyleft" terms (terms which require licensing other code under
+    compatible FOSS terms), such as kernel modules or plugins, then the
+    standard libraries for the target may potentially be subject to copyleft
+    terms, as long as such terms are satisfied by Rust's existing practices of
+    providing of full corresponding source code. Note that anything added to
+    the Rust repository itself must still use Rust's standard license terms.
+- Tier 2 targets must not impose burden on the authors of pull requests, or
+  other developers in the community, to ensure that tests pass for the target.
+  In particular, do not post comments (automated or manual) on a PR that
+  suggests a block on the PR based on tests failing for the target. (A PR
+  author must not break the build of a tier 2 target, but need not ensure the
+  tests pass for a tier 2 target, even if notified that they fail.)
+- The target maintainers should regularly run the testsuite for the target, and
+  should fix any test failures in a reasonably timely fashion.
+- All tier 3 requirements apply.
+
+A tier 2 target may be demoted or removed if it no longer meets these
+requirements. Any proposal for demotion or removal will be CCed to people who
+have previously worked on the target, and will be communicated widely to the
+Rust community before being dropped from a stable release.
+
+In some circumstances, especially if the target maintainers do not respond in a
+timely fashion, Rust teams may land pull requests that temporarily disable some
```

Comment:
Is there any expectation to try to give "sufficient time" for tier 2 maintainers to implement these features or is it fine for tier 1 support to land right before a new beta is cut (making it impossible for tier 2 maintainers to implement the feature)?